lib/elixir/lib/kernel/utils.ex

```diff
@@ -108,7 +108,18 @@ defmodule Kernel.Utils do
     :lists.foreach(foreach, enforce_keys)
 
     struct = :maps.put(:__struct__, module, :maps.from_list(fields))
-    {struct, enforce_keys, Module.get_attribute(module, :derive)}
+
+    case enforce_keys -- :maps.keys(struct) do
+      [] ->
+        {struct, enforce_keys, Module.get_attribute(module, :derive)}
+
+      error_keys ->
+        raise CompileError,
+          description:
+            "@enforce_keys required key/s (#{inspect(error_keys)}) that are not defined in defstruct: #{
+              inspect(fields)
+            }"
```

Comment:
```suggestion
        raise ArgumentError,
              "@enforce_keys required keys (#{inspect(error_keys)}) that are not defined in defstruct: " <>
                "#{inspect(fields)"
```

Changes:

1. CompileError is typically raised only by the compiler and when it has line+column info. ArgumentError is more appropriate here

2. Format the error message to less lines

3. Remove "key/s" idiom which is not used anywhere